src/devices/src/virtio/vsock/packet.rs

```diff
@@ -97,16 +97,23 @@ pub struct VsockPacket {
     buf_size: usize,
 }
 
-fn get_host_address(
-    mem: &GuestMemoryMmap,
+fn get_host_address<T: GuestMemory>(
+    mem: &T,
```

Comment:
I think making this generic is overkill. Because MmapRegion implements `VolatileMemory`, keeping `GuestMemoryMmap` would allow you to replace this method in the future with one returning a `VolatileSlice` (via `region.get_slice` instead of `region.get_host_address`). `VolatileSlice` is range-checked and thus avoids the well-known pitfalls of `get_host_address`.

Reply:
See also rust-vmm/vm-memory#76 if you want to keep the generic implementation.

Reply:
In the future we'll want to make everything generic anyway: #1544

So I think it would make sense to keep this as it is.

Reply:
Ok, I created rust-vmm/vm-memory#76 in order to add `get_slice` to the traits instead.